### ISSUE TITLE
Unable to delete or give user a `superuser` or `admin` roles with UI

### ISSUE THREAD

**Issue by SavchenkoAlexander** [getsentry/self-hosted#1288]
### Version

21.11-22.1

### Steps to Reproduce

1. Setup a self hosted instance;
2. Log in into this instance with first user (with id 1);
3. Then Invite and register a new user (with id 2);
4. Then go to `yourinstance.com/manage/users/2/`;
5. Try to check `Admin` or/and `Superuser` switch or Remove the user;

### Expected Result

User can be made a `Superuser` or/and `Admin` successfully. Also user can be removed.

### Actual Result

- If you try to make the user an `Admin` or `Superuser` there will be a notification about successful update but there is no changes on page update.
- If you try to remove the user UI doesn't react but web request returns a `403` code.

**Comment by aminvakil** [comment]
Can you check your nginx and/or your loadbalancer `error.log` and paste some more logs?

**Comment by SavchenkoAlexander** [comment]
I reproduced this issue on a new VM.

Tried both problems: removing user and making them Admin and Superuser
Then called `docker logs sentry-self-hosted_nginx_1> access.log 2> error.logs`. `error.logs` is empty both times.

Only the first problem leads to this line in `access.log`:
```
<client-ip> - - [03/Feb/2022:12:34:57 +0000] "DELETE /api/0/users/2/ HTTP/1.1" 403 64 "http://<sentry-ip>.18:9000/manage/users/2/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/97.0.4692.99 Safari/537.36" "-"
```
The second one produces successful PUT in `access.log`:
```
<client-ip> - - [03/Feb/2022:12:36:03 +0000] "PUT /api/0/users/2/ HTTP/1.1" 200 657 "http://<sentry-ip>:9000/manage/users/2/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/97.0.4692.99 Safari/537.36" "-"
```

**Comment by boskiv** [comment]
Same here. Can't give SSO user a superuser or admin permissions

**Comment by boskiv** [comment]
```
web_1                                       | 01:53:59 [INFO] sentry.superuser: superuser.request (url='http://stg-sentry.xxxxxx.io/api/0/users/2/' method='PUT' ip_address='10.176.35.84' user_id=1)
web_1                                       | 01:53:59 [INFO] sentry.audit.user: user.edit (user_id=2 actor_id=1 form_data={'id': '2', 'name': 'xxxxx xxxx', 'username': '9582d9eb0419495381875bd529b27d32', 'email': 'xxxxx@xxxxxxx.net', 'avatarUrl': 'https://secure.gravatar.com/avatar/d1e321efda5a1bd8177fc7fcb80f6ff3?s=32&d=mm', 'isActive': True, 'hasPasswordAuth': False, 'isManaged': False, 'dateJoined': '2022-02-04T01:52:52.585285Z', 'lastLogin': '2022-02-04T01:52:53.151716Z', 'has2fa': False, 'lastActive': '2022-02-04T01:52:52.585294Z', 'isSuperuser': False, 'isStaff': True, 'experiments': {}, 'avatar': {'avatarType': 'letter_avatar', 'avatarUuid': None}, 'identities': [{'id': '2', 'name': 'xxxxxx', 'organization': {'slug': 'sentry', 'name': 'Sentry'}, 'provider': {'id': 'active-directory', 'name': 'Active Directory'}, 'dateSynced': '2022-02-04T01:52:52.599751Z', 'dateVerified': '2022-02-04T01:52:52.599744Z'}], 'emails': [{'id': '2', 'email': 'xxxxx@xxxxxxx.net', 'is_verified': False}], 'permissions': [], 'authenticators': [], 'canReset2fa': True})
web_1                                       | 01:53:59 [INFO] sentry.access.api: api.access (method='PUT' view='sentry.api.endpoints.user_details.UserDetailsEndpoint' response=200 user_id='1' is_app='None' token_type='NoneType' organization_id='None' auth_id='None' path='/api/0/users/2/' caller_ip='10.176.35.84' user_agent='Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/97.0.4692.99 Safari/537.36' rate_limited='False' request_duration_seconds=0.030556917190551758)
nginx_1                                     | 192.168.128.1 - - [04/Feb/2022:01:53:59 +0000] "PUT /api/0/users/2/ HTTP/1.0" 200 882 "https://stg-sentry.xxxxx.io/manage/users/2/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/97.0.4692.99 Safari/537.36" "10.176.35.84"
nginx_1                                     | 192.168.128.1 - - [04/Feb/2022:01:53:59 +0000] "POST /api/31/store/ HTTP/1.0" 200 41 "-" "sentry.python/0.9.5" "10.226.80.167"
web_1                                       | 01:53:59 [INFO] sentry.access.api: api.access (method='POST' view='sentry.api.endpoints.relay_projectconfigs.RelayProjectConfigsEndpoint' response=200 user_id='None' is_app='None' token_type='NoneType' organization_id='None' auth_id='None' path='/api/0/relays/projectconfigs/' caller_ip='192.168.128.29' user_agent='None' rate_limited='False' request_duration_seconds=0.020581960678100586)
```

**Comment by chadwhitacre** [comment]
> P.S. Just setup a new 21.10.0 and there is no such issue. So something happends after

Interesting. I've dropped a line internally to get some eyes on this, let's see what we can figure out ...

**Comment by maxiuyuan** [comment]
Hi! This is happening since the first user doesn't have the correct permissions. We're looking for a way to fix the permission being added correctly but for now to fix this, use this command line tool: `sentry permissions add -u <USER_EMAIL> -p "users.admin"

**Comment by Leobaillard** [comment]
@maxiuyuan Thanks for the solution! I'm very new to Sentry and I was wondering where we should input this command? I have the sentry-cli but it appears that it does not handle permissions.

**Comment by skudnoff** [comment]
@maxiuyuan it works, thank you.

@Leobaillard try this
```bash
docker exec -ti sentry-self-hosted_web_1 sentry permissions add -u <USER_EMAIL> -p "users.admin"
```

**Comment by Leobaillard** [comment]
@skudnoff Thanks, it makes sense :) I executed the command but I still don't see any changes. The "admin" menu does not appear for the user I modified, even after reconnecting. I also can't see any changes in the permission toggles for that user using an admin user. Should I restart something?

**Comment by skudnoff** [comment]
@Leobaillard Giving user said permission doesn't make this user an admin, it fixes one specific problem described in the first post. Namely it fixes inability of admin user `A` to successfully save changes that make user `B` an admin by means of Sentry UI. User `A` have to be an admin already, and workaround suggests permission `users.admin` should be given to user `A`.

**Comment by Leobaillard** [comment]
Ok, I get it now, thanks a lot, it worked :)

**Comment by aminvakil** [comment]
@SavchenkoAlexander @boskiv Can you please check if the provided solution works for your too or not?

**Comment by SavchenkoAlexander** [comment]
@aminvakil Yes, the solution suggested by @maxiuyuan worked for me too.

**Comment by dcramer** [comment]
(duplicating my comment on the PR)

a f/u we might consider: creating (automatically) a Role that contains all known permissions, and syncing it somehow on migrate/update.

then self-hosted can just choose to grant this role vs a specific permission, which would be the equiv of the old superuser behavior. could be called "Super Admin" or something like that.

**Comment by aminvakil** [comment]
Ok, to summarize this, https://github.com/getsentry/sentry/pull/31714 has been merged and included in 22.2.0, and there is a workaround for affected users using older versions:

`sentry permissions add -u <USER_EMAIL> -p "users.admin"`

Let's close this then!
